pkg/etw/wrapper_64.go

```diff
@@ -46,7 +46,7 @@ func eventSetInformation(
 // for provider notifications. Because Go has trouble with callback arguments of
 // different size, it has only pointer-sized arguments, which are then cast to
 // the appropriate types when calling providerCallback.
-func providerCallbackAdapter(sourceID *guid.GUID, state uint32, level uint8, matchAnyKeyword uintptr, matchAllKeyword uintptr, filterData uintptr, i uintptr) uintptr {
```

Comment:
I'm a bit confused as to why we need to change these types. From [PENABLECALLBACK](https://docs.microsoft.com/en-us/windows/win32/api/evntprov/nc-evntprov-penablecallback) it looks like they are `ULONG` and `UCHAR`, which should correspond to `uint32` and `uint8`.

Reply:
So this is actually reverting the types to what they were before my initial changes. The comment above explains why they were that way. Go appears to give us all the callback variables as pointers, so if you leave level as a uint8, then part of the level gets shoved into matchAnyKeyword and everything becomes misaligned, causing the corruption.

Reply:
Ah, I think I see now, this is presumably something to do with how Go's callback code is translating the native args to the Go ABI?